.github/workflows/pre-commit.yaml

```diff
@@ -0,0 +1,30 @@
+name: Pre-commit linters and code checks
+
+on:
+  pull_request:
+    paths-ignore:
+      - 'README.md'
+  push:
+    branches:
+      - develop
+    paths-ignore:
+      - 'README.md'
+
+jobs:
+  pre_commit:
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Set up Python
+      uses: actions/setup-python@v3
+      with:
+        python-version: 3.8
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pre-commit==3.3.3
+
+    - name: Run pre-commit
+      run: pre-commit run
```

Comment:
Do you think it will run? Don't you need to install pre-commit hooks first?
https://github.com/OML-Team/open-metric-learning/blob/9e89e1bd7ee13371918f0edff66c18ea3fe587fe/Makefile#L73

Reply:
It seems to run according to my tests. I was inspired by
https://github.com/OML-Team/open-metric-learning/blob/main/.github/workflows/pre-commit-workflow.yaml

Reply:
I'm just wondering if this will work without `pre-commit install` called first

Reply:
We can run with pre-commit run --all-files in the next PR to isolate the work.